doc/plutip-testing.md

```diff
@@ -92,34 +123,52 @@ runPlutipContract config distribution \wallets -> do
 
 In most cases at least two UTxOs per wallet are needed (one of which will be used as collateral, so it should exceed `5_000_000` Lovelace).
 
-### Testing with Mote
 
-`Contract.Test.Plutip.testPlutipContracts` type is defined as follows:
+Internally `runPlutipContract` runs a contract in an `Aff.bracket`, which creates Plutip cluster during the setup and terminates it during the teardown or in the case of an exception.
+Logs will be printed in case of error.
+
+### Testing with Mote
 
+`Contract.Test.Plutip.testPlutipContracts` type is defined as follows (after expansion of the CTL's `TestPlanM` type synonym):
 ```purescript
+type TestPlanM :: Type -> Type -> Type
+type TestPlanM test a = MoteT Aff test Aff a
+
 testPlutipContracts
   :: PlutipConfig
-  -> MoteT Aff PlutipTest Aff Unit
+  -> MoteT Aff ContractTest Aff Unit
   -> MoteT Aff (Aff Unit) Aff Unit
-```
-
-The final `MoteT` type requires the bracket, test and test building type to all be in `Aff`. The brackets cannot be ignored in the `MoteT` test runner, as it is what allows a single plutip instance to persist over multiple tests.
 
-To create tests of type `PlutipTest`, the user should either use `Contract.Test.Plutip.withWallets` or `Contract.Test.Plutip.noWallet`:
+-- Recall that `MoteT` has three type variables
+newtype MoteT bracket test m a
+```
+where
+* `bracket :: Type -> Type` is where brackets will be run (before/setup is `bracket r` and after/teardown is of type `r -> bracket Unit`),
+   * in our case it's `Aff` and is where the CTL environment and Plutip cluster setup will happen,
+   * also environment setup and Plutip startup and teardown will happen once per each top-level test or group inside the `testPlutipContracts` call,
+   * so wrap your tests or groups in a single group if you want for the cluster to start only once,
+* `test :: Type` is a type of tests themselves,
+   * in our case it's [`ContractTest`](../src/Internal/Test/ContractTest.purs), which in a nutshell describes a function from some wallet UTxO distribution to a `Contract r`
+   * wallet UTxO distribution is the one that you need to pattern-match on when writing tests
+* `m :: Type -> Type` is a monad where effects during the construction of the test suite can be performed,
+   * here we use `Aff` again
+* `a :: Type` is a result of the test suite, we use `Unit` here.
```

Comment:
This whole section should me de-prioritized. We must first tell the user how to use `withWallets`, and only then dive into the techinical detail of an internal type. Move it to a separate sub-section and state at the beginning of the paragraph that this is an explanation how Mote and our intergation with it works *internally*

Reply:
Moved it to a separate section, combined with other internal notes